src/hooks/admin/use-localization-toolkit/useLocalizationToolkit.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from 'react';
+import { LocalizationLanguage, TranslationStatusFilter } from '../../../types/common/language';
+import { localizationLanguagesDataFetch } from '../../../services/api/public/localization/languages/languages-api';
+import axios from 'axios';
+import { COMMON_TEXT_ADMIN } from '../../../const/admin/common';
+import { DEFAULT_LOCALE } from '../../../const/common/locales';
+
+export interface UseLocalizationToolkitProps {
+    setErrorState: (message: string, type: any) => void;
+}
+
+export function useLocalizationToolkit({ setErrorState }: UseLocalizationToolkitProps) {
+    const [allLanguages, setAllLanguages] = useState<LocalizationLanguage[]>([]);
+    const [translationLanguages, setTranslationLanguages] = useState<LocalizationLanguage[]>([]);
+    const [selectedLanguage, setSelectedLanguage] = useState<LocalizationLanguage>();
+    const [translationStatusFilter, setTranslationStatusFilter] = useState<TranslationStatusFilter | undefined>();
```

Comment:
I believe it can be refactor to avoid derived state
```suggestion
    const [allLanguages, setAllLanguages] = useState<LocalizationLanguage[]>([]);
    const [selectedLanguage, setSelectedLanguage] = useState<LocalizationLanguage | null>(null);
    const [translationStatusFilter, setTranslationStatusFilter] = useState<TranslationStatusFilter>(
        TranslationStatusFilter.All
    );
    
    const translationLanguages = useMemo(() => {
        return allLanguages.filter((language) => language.code !== DEFAULT_LOCALE);
    }, [allLanguages]);
```

Reply:
we can`t implement this suggestion because our main language is ukrainian and other foreign languages we need to deliver for other component to set his badge or badges, so we don`t need to change it. I mean we can`t remove translationLanguages state.